### ISSUE TITLE
Return HTTP 404 rather than 405 for non-existent methods

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#3112]
## Description

Presently Lighthouse's HTTP API will return a `405 Method Not Allowed` for non-existent endpoints. I believe this behaviour is incorrect semantically, as well inconsistent with other clients.

Semantically, 405 means we received a request with a valid path, but an invalid method (like `GET`, `POST`, etc). Per: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

When sending a 405 we should be sending an `Allow` header back (we do not), but that also doesn't make sense for URLs we don't recognise. I think the correct behaviour would be to return a 404 instead.

## Version

Lighthouse v2.1.5

## Present Behaviour
```
$ curl -v "http://localhost:5052/not_real"
*   Trying 127.0.0.1:7052...
* Connected to localhost (127.0.0.1) port 7052 (#0)
> GET /not_real HTTP/1.1
> Host: localhost:7052
> User-Agent: curl/7.74.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 405 Method Not Allowed
< content-type: application/json
< server: Lighthouse/v2.1.5-ffcfef0/x86_64-linux
< content-length: 60
< date: Wed, 23 Mar 2022 03:59:49 GMT
< 
* Connection #0 to host localhost left intact
{"code":405,"message":"METHOD_NOT_ALLOWED","stacktraces":[]}
```

## Steps to resolve

Untangle the `warp` filters such that we return a 404. There seems to be an issue with 405s taking priority over 404s: https://github.com/seanmonstar/warp/issues/77

**Comment by santiagomed** [comment]
Hi, I'd love to help with this if still needed. I'll start looking into it! I'd also appreciate it if someone were to point me in the right direction since I just started getting familiar with the codebase.

**Comment by michaelsproul** [comment]
Thanks @santiagomed that would be great! The place to start is probably in `beacon_node/http_api/src/lib.rs`. That's where we build up the big chain of warp routes and filters.

**Comment by santiagomed** [comment]
Created a PR addressing this: #3836.

**Comment by michaelsproul** [comment]
Closed by #3836! 🎉 

Thanks @santiagomed
